src/main/java/me/karakelley/http/server/HttpRequestReader.java

```diff
@@ -78,8 +81,13 @@ private String readRequestLineAndHeaders() throws IOException {
     return buffer.toByteArray();
   }
 
-  private String validPath(String path) {
-    if (path == null) throw new InvalidRequestException("Invalid Path!");
+  private String validPath(String path) throws AttemptedDirectoryTraversalException {
+    Path userPath = Paths.get(path);
+
+    Path normalizedPath = userPath.normalize();
+    if (normalizedPath != userPath || userPath.startsWith("..")) {
```

Comment:
Yes, that's how `normalize` works - So once the path has been normalized and it's different that what you requested then it'll throw an `AttemptedDirectoryTraversalException`. 

So if you requested `/somedirectory/../../.ssh` it'll be normalized to /somedirectory/.ssh. Then we check to see if `/somedirectory/../../.ssh` is equal to ` /somedirectory/.ssh`. They're not equal so exception is thrown. 

I believe checking either `userPath` or `normalizedPath` for preceding dots will the same since `normalize()` doesn't remove any preceding dots however it does seem more readable to check the `normalizedPath` instead.